Fix mobile nav rendering

mitchellhenke opened the pull request:
from @aduth's comment at: https://github.com/18F/identity-idp/pull/4169#pullrequestreview-489740982

aduth reviewed (approved):
Two observations:

- In how this is implemented on the USWDS documentation site, they are rendering the mobile navigation element as a direct child of the `body` tag. Since we're styling the container as `position: fixed`, it might not be an issue, though I have seen situations previously where styles could cascade unintentionally from a parent element into a modal dialog, which is less likely to happen if the ancestry is shorter.
- It's very subtle, but I notice a slight difference in the padding of the desktop header after these changes. This may also be related to how it previously wasn't rendered in standards mode (doctype html).

Before|After
---|---
![before](https://user-images.githubusercontent.com/1779930/93370926-8bb83000-f81f-11ea-88c7-24429743f4e0.png)|![after](https://user-images.githubusercontent.com/1779930/93370907-8529b880-f81f-11ea-8cc4-1b749aae2dd3.png)

Not in scope here, but another related question: Should the background be dimmed when the mobile menu is opened? I think this requires that a `<div class="usa-overlay"></div>` be present on the page.

mitchellhenke commented:
@aduth interestingly, prod/staging have the taller navbar, #4169 caused it to narrow, and now these changes are ending up causing it to go back to "normal"

I've moved the mobile nav to be a `content_for` so it can be rendered as a direct child of the body, and added a `usa-overlay` to the body as well.

zachmargolis reviewed (approved):
Should we add a spec to make sure this doesn't regress? That the elements are nested in the right order? Unsure how useful/annoying it would be

aduth commented on app/views/layouts/base.html.erb:
```diff
@@ -54,6 +54,7 @@
 </head>
 
 <body class="<%= Rails.env %>-env site <%= yield(:background_cls) %>">
+  <%= content_for?(:mobile_nav) ? yield(:mobile_nav) : yield %>
```
Based on the build failures, I think the "else" condition here is causing the page content to be double-printed.

```suggestion
  <%= yield(:mobile_nav) if content_for?(:mobile_nav) %>
```

aduth commented on app/views/accounts/connected_accounts/show.html.erb:
```diff
@@ -1,3 +1,5 @@
+<% title t('account.navigation.connected_accounts') %>
+
 <h1 class="mt0">
   <%= t('headings.account.connected_accounts') %>
```
Thinking out loud a bit: It's sensible that the page's title and the page's `h1` are the same, which raises a question whether it's worth having duplicate strings for these. And if we do have duplicate strings (one for the navigation and one for the page title), whether the value we use in `title` should be the same as the page's `h1`, or the same as what's shown in the navigation.

mitchellhenke replied:
Yeah, 100% agree
